sdk/storage/Azure.Storage.Blobs.Batch/src/BlobBatch.cs

```diff
@@ -153,15 +191,36 @@ private void SetBatchOperationType(BlobBatchOperationType operationType)
             DeleteSnapshotsOption snapshotsOption = default,
             BlobRequestConditions conditions = default)
         {
-            var blobUri = new BlobUriBuilder(_client.Uri)
-            {
-                BlobContainerName = blobContainerName,
-                BlobName = blobName
-            };
-            return DeleteBlob(
-                blobUri.ToUri(),
-                snapshotsOption,
-                conditions);
+            SetBatchOperationType(BlobBatchOperationType.Delete);
+
+            HttpMessage message = BlobRestClient.CreateDeleteRequest(
+                containerName: blobContainerName,
+                blob: blobName,
+                timeout: null,
+                leaseId: conditions?.LeaseId,
+                deleteSnapshots: snapshotsOption == DeleteSnapshotsOption.None ? null : (DeleteSnapshotsOptionType?)snapshotsOption,
+                ifModifiedSince: conditions?.IfModifiedSince,
+                ifUnmodifiedSince: conditions?.IfUnmodifiedSince,
+                ifMatch: conditions?.IfMatch.ToString(),
+                ifNoneMatch: conditions?.IfNoneMatch.ToString(),
+                ifTags: conditions?.TagConditions,
+                blobDeleteType: null);
+
+            _messages.Add(message);
+
+            return new DelayedResponse(
+                message,
+                async response =>
+                {
+                    switch (response.Status)
+                    {
+                        case 202:
+                            BlobDeleteHeaders blobDeleteHeaders = new BlobDeleteHeaders(response);
+                            return ResponseWithHeaders.FromValue(blobDeleteHeaders, response);
+                        default:
+                            throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false);
```

Comment:
when this code is executed? would it make public API throw or would errored message be in enumeration in the API result?
Worth checking if we have existing test that hits such logical branch. If we have one then we're good, if we don't we should get one.
(similar question elsewhere, it seems to form a pattern)

Reply:
This code is executed when a sub-request fails, an actual exception is not thrown, the error message is added to the result.

There are multiple unit tests (basically any unit tests with the word "fail" in the name) that cover this path, hence how I found I needed to add this.